src/main/java/io/dexi/config/Config.java

```diff
@@ -0,0 +1,187 @@
+package io.dexi.config;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.FileBasedConfiguration;
+import org.apache.commons.configuration2.YAMLConfiguration;
+import org.apache.commons.configuration2.builder.FileBasedConfigurationBuilder;
+import org.apache.commons.configuration2.builder.ReloadingFileBasedConfigurationBuilder;
+import org.apache.commons.configuration2.builder.fluent.Parameters;
+import org.apache.commons.configuration2.builder.fluent.PropertiesBuilderParameters;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.commons.lang.StringUtils;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ *  This class supports reading a "hierarchical" configuration file with sections like:
+ *
+ *  <b>YAML example</b>
+ *  <pre>
+ *  {@code
+ *     dexi:
+ *       baseUrl: http://localhost:3000/api/
+ *       apiKey: super-secret-key
+ *  }
+ *  </pre>
+ *
+ *  <b>JSON example</b>
+ *  <pre>
+ *  {@code
+ *      {
+ *          "dexi": {
+ *              "baseUrl": "http://localhost:3000/api/",
+ *              "apiKey": "super-secret-key"
+ *          }
+ *      }
+ *  }
+ *  </pre>
+ *
+ *  The file can be read from a local disk or from a URL. The class also supports reading environment variables whose
+ *  names start with "DEXI_APP_".
+ *
+ *  Configuration is read in the following order:
+ *  <ol>
+ *      <li>A local configuration file as specified by the {@code localConfigFile} parameter. The default is
+ *          {@code ~/.dexi/my-app.yml}.
+ *          <ul>
+ *              <li>Supported file formats are YAML (.yml), JSON (.json), XML (.xml) and INI (.ini).</li>
+ *          </ul>
+ *      </li>
+ *      <li>If an environment variable or system property named {@code DEXI_APP_CONFIG_URL_YML} is set, read a YAML (.yml)
+ *          file from the specified URL.</li>
+ *      <li>Read any {@code DEXI_APP_} environment variable. The format is: {@code DEXI_APP_&lt;section>_&lt;key> = &lt;value>}.</li>
+ *  </ol>
+ *
+ *  Values for duplicate keys within sections are overwritten by later keys.
+ *
+ */
+public class Config {
+
+    public static final String DEXI_APP_CONFIG_URL = "DEXI_APP_CONFIG_URL";
```

Comment:
I think this should be DEXI_APP_CREDENTIALS (To mimic google a bit) - and the value can be either a URL or a file

Reply:
Ahh ok, so you only see this file as holding credentials? That explains your "missing account id" comment above.

What I have implemented is a general utility to read a configuration file.

Reply:
Well I guess it makes sense to allow other values in here - so maybe make it DEXI_APP_CONFIG then ?